modules/ossm-installing-gateway-using-gateway-injection.adoc

```diff
@@ -0,0 +1,255 @@
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: secret-reader
+  namespace: <gateway_namespace>
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: secret-reader
+  namespace: <gateway_namespace>
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name:  secret-reader
+  namespace: <gateway_namespace>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: secret-reader
+subjects:
+  - kind: ServiceAccount
+    name:  secret-reader
+----
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f secret-reader.yml
+----
+
+. Create a YAML file named `gateway-deployment.yml` that defines the Kubernetes `Deployment` for the gateway. 
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: <gateway_name>
+  namespace: <gateway_namespace>
+spec:
+  selector:
+    matchLabels:
+      istio: <gateway_name> 
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway <1>
+      labels:
+        istio: <gateway_name> <2>
+        sidecar.istio.io/inject: "true" <3>
+    spec:
+      containers:
+        - name: istio-proxy
+          image: auto <4>
+          securityContext:
+            capabilities:
+              drop:
+              - ALL
+            allowPrivilegeEscalation: false
+            privileged: false
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+          ports:
+          - containerPort: 15090
+            protocol: TCP
+            name: http-envoy-prom
+          resources:
+            limits:
+              cpu: 2000m
+              memory: 1024Mi
+            requests:
+              cpu: 100m
+              memory: 128Mi
+      securityContext:
+        sysctls:
+        - name: net.ipv4.ip_unprivileged_port_start
+          value: "0"
+      serviceAccountName: secret-reader <5>
+----
+<1> Indicates that the {istio} control plane uses the gateway injection template instead of the default sidecar template.
+<2> Ensure that a unique label is set for the gateway deployment. A unique label is required so that {istio} `Gateway` resources can select gateway workloads. 
+<3> Enables gateway injection by setting the `sidecar.istio.io/inject` label to `true`. If the name of the {istio} resource is not `default` you must use the `istio.io/rev: <istio_revision>` label, where the revision represents the active revision of the {istio} resource.
+<4> Sets the image field to `auto` so that the image automatically updates each time the pod starts.
+<5> Sets the `serviceAccountName` to the name of the `ServiceAccount` created in the previous step.
```

Comment:
"previous step" is no longer accurate as the step being referenced is now two steps before this.

```suggestion
<5> Sets the `serviceAccountName` to the name of the `ServiceAccount` previously created.
```

Reply:
done